util/src/main/java/io/kubernetes/client/util/ClientBuilder.java

```diff
@@ -12,231 +12,217 @@
-		}
-
-		if(password != null){
-			client.setPassword(password);
-		}
-
-		if(( userName != null )&&(password != null)) {
-			final String usernameAndPassword = userName + ":" + password;
-			client.setApiKeyPrefix("Basic");
-			client.setApiKey(ByteString.of(usernameAndPassword.getBytes(Charset.forName("ISO-8859-1"))).base64());
-		}
-
-		if(accessToken != null) {
-			if (apiKeyPrefix == null){
-				client.setApiKeyPrefix("Bearer");
-			}
-			client.setAccessToken(accessToken);
-		}
-
-		if(apiKeyPrefix != null) {
-			client.setApiKeyPrefix(apiKeyPrefix);
-		}
-
-		if(apiKey != null) {
-			if (apiKeyPrefix == null){
-				client.setApiKeyPrefix("Bearer");
-			}
-			client.setApiKey(apiKey);
-		}
-
-		client.setVerifyingSsl(verifyingSsl);
-
-		if(certificateAuthorityFile != null) {
-			client.setSslCaCert(new FileInputStream(certificateAuthorityFile));
-		}
-
-		if(certificateAuthorityData != null) {
-			byte[] bytes = Base64.decodeBase64(certificateAuthorityData);
-			client.setSslCaCert(new ByteArrayInputStream(bytes));
-		}
-
-		return client;
-	}
+  private static final Logger log = Logger.getLogger(ClientBuilder.class);
+
+  private String basePath = Config.DEFAULT_FALLBACK_HOST;
+  private byte[] caCertBytes = null;
+  private boolean verifyingSsl = true;
+  private Authentication authentication;
+
+  /**
+   * Creates an {@link ApiClient} by calling {@link #standard()} and {@link #build()}.
+   *
+   * @return An <tt>ApiClient</tt> configured using the precedence specified for {@link #standard()}.
+   * @throws IOException
+   *  if the configuration file or a file specified in a configuration file cannot be read.
+   */
+  public static ApiClient defaultClient() throws IOException {
+    return ClientBuilder.standard().build();
+  }
+
+  /**
+   * Creates a builder which is pre-configured in the following way
+   *
+   * <ul>
+   *   <li>If $KUBECONFIG is defined, use that config file.</li>
+   *   <li>If $HOME/.kube/config can be found, use that.</li>
+   *   <li>If the in-cluster service account can be found, assume in cluster config.</li>
+   *   <li>Default to localhost:8080 as a last resort.</li>
+   * </ul>
+   *
+   * @return <tt>ClientBuilder</tt> pre-configured using the above precedence
+   * @throws IOException
+   *  if the configuration file or a file specified in a configuration file cannot be read.
+   */
+  public static ClientBuilder standard() throws IOException {
+    final FileReader kubeConfigReader = findConfigFromEnv();
+    if(kubeConfigReader != null) {
+      return kubeconfig(loadKubeConfig(kubeConfigReader));
+    }
+    final FileReader configReader = findConfigInHomeDir();
+    if(configReader != null) {
+      return kubeconfig(loadKubeConfig(configReader));
+    }
+    final File clusterCa = new File(SERVICEACCOUNT_CA_PATH);
+    if (clusterCa.exists()) {
+      return cluster();
+    }
+    return new ClientBuilder();
+  }
+
+  private static FileReader findConfigFromEnv() {
+    try {
```

Comment:
Given that the only caller that calls `findConfigFromEnv` throws IOException, let's just mark this method as throwing `FileNotFoundException` and throw it through to the caller?